### PR TITLE
JAMES-4059 MailetContainerModule should not do a strict check on RemoveMimeHeader with bcc name param

### PR DESCRIPTION
**Pull request by Arsnael**
I hesitated adding a test for the jvm property commit as it's a bit tricky for james startup

Maybe in memory-app with some alternative conf files in resources? Or just not necessary?

### CODE REVIEW

File: docs/modules/servers/pages/distributed/configure/jvm.adoc [hunk 1]
```diff
@@ -103,3 +103,18 @@ Ex in `jvm.properties`
 # james.reactor.inputstream.prefetch=4
 ----
 
+== Disable mailet container check at James startup
+
+James is doing a check on startup making sure that the `bcc` header is being removed via
+`RemoveMimeHeader` mailet in the mail processing pipeline defined in `mailetcontainer.xml` file.
+
+It could be useful for some administrators that know what they are doing to disable such a check
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/configure/jvm.adoc:111]:
```suggestion
It could be useful for some administrators that know what they are doing to disable such checks
```

File: server/container/guice/protocols/jmap/src/test/java/org/apache/james/jmap/MailetPreconditionTest.java [hunk 2]
```diff
@@ -191,5 +191,19 @@ void bccMailetCheckShouldNotThrowOnValidPair() throws Exception {
             assertThatCode(() -> MailetContainerModule.BCC_Check.check(pairs))
                 .doesNotThrowAnyException();
         }
+
+        @Test
+        void bccMailetCheckShouldNotThrowOnValidPairWithOtherNames() throws Exception {
```
Comment (chibenwa) [inline server/container/guice/protocols/jmap/src/test/java/org/apache/james/jmap/MailetPreconditionTest.java:196]: Edge case: Please add a test ensuring that removing `x-bcc-bug` header is not considered as valid.

**Review by chibenwa — COMMENTED** [review]
While we are at it, how about case insentivity?

(RemoveMimeHeader is case insentive, the check here is case sensitive)

**Review by quantranhong1999 — APPROVED** [review]

File: server/container/guice/mailet/src/main/java/org/apache/james/modules/server/MailetContainerModule.java [hunk 3]
```diff
@@ -86,7 +86,7 @@ public class MailetContainerModule extends AbstractModule {
         "transport",
         All.class,
         RemoveMimeHeader.class,
-        pair -> Arrays.asList(pair.getMailet().getMailetConfig().getInitParameter("name").split(",")).contains("bcc"),
+        pair -> Arrays.asList(pair.getMailet().getMailetConfig().getInitParameter("name").toLowerCase().split(",")).contains("bcc"),
```
Comment (chibenwa) [inline server/container/guice/mailet/src/main/java/org/apache/james/modules/server/MailetContainerModule.java:89]: Other stupid edge case question: do we handle the case where RemoveMimeHeader do not have a "name" property decently?

IE we likely throw today an uggly non explicit NPE, it would be nice to throw a contextualized with decent error message IllegalStateException, and have a unit test for it?

Comment (chibenwa) [reply server/container/guice/mailet/src/main/java/org/apache/james/modules/server/MailetContainerModule.java:89]: Likely OK

**Review by chibenwa — APPROVED** [review]
